marker-api.js

```diff
@@ -1,780 +1,780 @@
-    },
-
-    errsByCode: {
-        "17287": "There is a problem with your coordinates. Please make sure all four coordinates are valid floating point numbers"
-    }
-}
+      }
+
+          // Build return data
+          returnData = marker.toObject();
+          returnData = _.omit(returnData, 'user_id', '__v');
+
+          res.status(201).json({ 
+              message: 'New marker save successful',
+              data: returnData
+          });
+      });
+  }
+
+  function handle_marker_data_failure () {
+      console.log('marker data failure');
+      return res.status(400).json({message: 'Error parsing form data. Please ensure form data is well formed.'});
+  }
+
+  /**
+   * Returns Q promise
+   */
+  function get_marker_data (marker_ids) {
+      var def = Q.defer(),
+          obj_ids = [];
+
+      if (!marker_ids || !marker_ids.length) return false;
+
+      // Cast to ObjectId's
+      obj_ids = marker_ids.map((id) => {
+          return ObjectId(id);
+      });
+
+      Marker
+          .aggregate([
+
+              // Find markers by id
+              { "$match" : {'_id': { '$in': obj_ids } } },
+
+              // Join username from users collection
+              { "$lookup" : { 
+                  "from" : "users",
+                  "localField" : "user_id",
+                  "foreignField" : "_id",
+                  "as" : "user_info" }
+              },
+
+              // Project (filter) only necessary fields
+              { "$project" : {
+          "createdDate" : 1,
+          "tags" : 1,
+          "photo_hash" : 1,
+          "geometry" : 1,
+          "user_info._id" : 1,
+          "approved": 1
+          }}
+          ])
+          .exec(function (err, markers) {
+              if (err)
+                  return def.reject({'message': 'An internal error occurred', 'status': 500});
+
+              if (!markers || !markers[0])
+                  return def.reject({'status': 404, message: 'No markers found with ids: ' + obj_ids});
+
+              // Build return data
+              returnData = markers;
+              returnData.forEach(function (marker) {
+
+                  // Make joined username field a property of main object
+                  if (marker.user_info && marker.user_info.length) {
+                      marker.user_id = marker.user_info[0]._id;
+                  }
+                  delete marker.user_info;
+              });
+
+              def.resolve(returnData);
+          });
+
+      return def.promise;
+  }
+
+  /**
+   * Get base 64 image data for a photo by it's id an size
+   * @param markers {object} key ->
+   */
+  function get_photo_b64 (markers) {
+      var def = Q.defer(),
+          sizes,
+          size_suffix,
+          b64_data = {},
+          reads = [],
+          ids;
+
+          console.log('photo requests: ' + JSON.stringify(markers));
+
+      if (!markers) {
+          console.log('invalid data for get_photo_b64');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/charlieblevins/dukweb/pullRequest?prid=9134160)
```suggestion
          console.log("invalid data for get_photo_b64");
```